factory/usr/lib/systemd/system-generators/kernel-snap-generator

```diff
@@ -0,0 +1,57 @@
+#!/bin/sh
+
+set -eu
+
+: "${KERNEL_MNT_POINT:=/run/mnt/kernel}"
+
+if ! mountpoint -q "${KERNEL_MNT_POINT}"; then
+    exit 0
+fi
+
+# FIXME use SYSTEMD_IN_INITRD when using systemd 251+
+if [ -f /etc/initrd-release ]; then
+    sysroot=/sysroot
+    sysroot_unit=sysroot
+    target=initrd-fs.target
+else
+    sysroot=
+    sysroot_unit=
+    target=local-fs.target
+fi
+
+: "${OS_RELEASE:="${sysroot}/etc/os-release"}"
+
+# Note that /sysroot/etc/os-release is accessible when generators are
+# re-executed due to "daemon-reload" called by
+# initrd-parse-etc.service
+
+# We generate the kernel bind mount units only
+# if we are on Ubuntu Classic
+
+# Ubuntu Core will have those mounts declared in /etc/fstab, thus
+# using systemd-fstab-generator instead.
```

Comment:
Which `/etc/fstab` do we mean here? the one in the initrd or the one in the installed system?

If need be, we can and should modify classic system's `/etc/fstab` such that required mounts are listed there, and then drop this.